### PR TITLE
Add millisecond field for time map initialize.

### LINKED ISSUE #2612: Setting millisecond failed but microsecond worked
Nebula Graph version: 2.5.0

Why can't I set millisecond with time() but microsecond worked? See the code below.

```
(root@nebula) [basketballplayer]> WITH time({hour: 12, minute: 31, second: 14, millisecond: 222}) AS d RETURN d
+----------+
| d        |
+----------+
| BAD_DATA |
+----------+
Got 1 rows (time spent 295/696 us)

Fri, 20 Aug 2021 09:16:34 UTC

(root@nebula) [basketballplayer]> WITH time({hour: 12, minute: 31, second: 14, microsecond: 222}) AS d RETURN d
+-----------------+
| d               |
+-----------------+
| 12:31:14.000222 |
+-----------------+
Got 1 rows (time spent 347/722 us)
```

### PR DESCRIPTION
**Pull request by Shylock-Hg**
Close #2612

### CODE REVIEW

**Comment by yixinglu** [comment]
Please add some integration tests in TCK

**Comment by Shylock-Hg** [comment]
> Please add some integration tests in TCK

Done

File: src/common/time/TimeUtils.cpp [hunk 1]
```diff
@@ -55,11 +55,16 @@ constexpr int64_t kMaxTimestamp = std::numeric_limits<int64_t>::max() / 10000000
         return Status::Error("Invalid second number `%ld'.", kv.second.getInt());
       }
       dt.sec = kv.second.getInt();
+    } else if (kv.first == "millisecond") {
```
Comment (CPWstatic) [inline src/common/time/TimeUtils.cpp:58]: Why not use switch here?

File: tests/tck/features/mutate/InsertWithTimeType.feature [hunk 2]
```diff
@@ -111,9 +111,9 @@ Feature: Insert with time-dependent types
       """
       UPDATE VERTEX "test"
       SET
-        tag_date.f_date = Date("2018-03-04"),
-        tag_date.f_time = Time("22:01:00"),
-        tag_date.f_datetime = DateTime("2018-03-04T22:30:40")
+        tag_date.f_date = Date({year: 2018, month: 3, day: 4}),
+        tag_date.f_time = Time({hour: 22, minute: 1, second: 0, millisecond: 0, microsecond: 0}),
+        tag_date.f_datetime = DateTime({year: 2018, month: 3, day: 4, hour: 22, minute: 30, second: 40, millisecond: 0, microsecond: 0})
```
Comment (CPWstatic) [inline tests/tck/features/mutate/InsertWithTimeType.feature:116]: Why not add a new test case instead of modifying?

File: src/common/time/TimeUtils.cpp (hunk 1 above)
Comment (Shylock-Hg) [reply src/common/time/TimeUtils.cpp:58]: switch can't process string type.

File: tests/tck/features/mutate/InsertWithTimeType.feature (hunk 2 above)
Comment (Shylock-Hg) [reply tests/tck/features/mutate/InsertWithTimeType.feature:116]: The origin case is tested in other case too.

**Review by yixinglu — APPROVED** [review]

**Review by Aiee — APPROVED** [review]

File: src/common/time/TimeUtils.cpp [hunk 3]
```diff
@@ -55,11 +55,16 @@ constexpr int64_t kMaxTimestamp = std::numeric_limits<int64_t>::max() / 10000000
...
+      if (kv.second.getInt() < 0 || kv.second.getInt() > 999) {
+        return Status::Error("Invalid millisecond number `%ld'.", kv.second.getInt());
```
Comment (yixinglu) [inline src/common/time/TimeUtils.cpp:60]: Could you add more boundary value test cases? such as ms is 999 or 1000, us is 999 or 1000.

Comment (Shylock-Hg) [reply src/common/time/TimeUtils.cpp:60]: Good catch. Done.

**Review by yixinglu — APPROVED** [review]
LGTM

**Review by Aiee — APPROVED** [review]
